Documentation for pi4ioe5v9xxxx integration

antonverburg opened the pull request:
**Description:**


**Pull request in home-assistant (if applicable):** home-assistant/home-assistant#28847>

antonverburg committed:
Logo of diodes.com, for pi4ioe5v9xxxx devices 

Added integration for pi4ioe5v9xxxx devices of Diodes.com

antonverburg committed:
Create pi4ioe5v9xxxx.markdown

Add documentation for new integration pi4ioe5v9xxxx

klaasnicolaas commented on source/_integrations/pi4ioe5v9xxxx.markdown:
```diff
@@ -0,0 +1,139 @@
+  - DIY
+  - Binary Sensor
+  - Switch
+ha_release: 1.101
```
```suggestion
ha_release: 0.103
```

klaasnicolaas commented on source/_integrations/pi4ioe5v9xxxx.markdown:
````diff
@@ -0,0 +1,139 @@
+
+```yaml
+# Example configuration.yaml entry
+binary_sensor:
````
Only required config variables in the first example

klaasnicolaas commented on source/_integrations/pi4ioe5v9xxxx.markdown:
````diff
@@ -0,0 +1,139 @@
+
+```yaml
+# Example configuration.yaml entry
+switch:
````
same

antonverburg committed:
Update source/_integrations/pi4ioe5v9xxxx.markdown

Co-Authored-By: Klaas Schoute <klaas_schoute@hotmail.com>

antonverburg committed:
Removed not-required config parameters

mrand commented on source/_integrations/pi4ioe5v9xxxx.markdown:
```diff
@@ -0,0 +1,136 @@
+    "pin: name":
+      description: The pin numbers (from 1 to X) and corresponding names.
+      required: true
+      type: [integer, string]
```
Would a colon be clearer than a comma, since the format actually uses colon?  (both here in and in the binary_sensor)?  Seems like the example format would be more clear when written as: 

      type: [integer] : [string]

frenck replied:
@mrand That suggested type is invalid.

mrand commented on source/_integrations/pi4ioe5v9xxxx.markdown:
```diff
@@ -0,0 +1,136 @@
+      required: true
+      type: [integer, string]
+invert_logic:
+  description: If `true`, inverts the output logic to ACTIVE LOW.
```
```suggestion
  description: If `true`, inverts the input logic to ACTIVE LOW.
```

antonverburg committed:
Update source/_integrations/pi4ioe5v9xxxx.markdown

Co-Authored-By: Marc Randolph <mrand@users.noreply.github.com>

frenck commented on source/_integrations/pi4ioe5v9xxxx.markdown:
```diff
@@ -0,0 +1,136 @@
+---
+title: "pi4ioe5v9xxxx IO expander"
```
```suggestion
title: "pi4ioe5v9xxxx IO Expander"
```

frenck commented on source/_integrations/pi4ioe5v9xxxx.markdown:
```diff
@@ -0,0 +1,136 @@
+The `pi4ioe5v9xxxx` integration provides support for the quasi-bidirectional devices PI4IOE5V9570, PI4IOE5V9674, PI4IOE5V9673, PI4IOE5V96224 and PI4IOE5V96248 from digital.com.
+
+For more details about the pi4ioe5v9xxxx I2C I/O port expander you can find the datasheets here:
+[PI4IOE5V9570](https://www.diodes.com/assets/Datasheets/PI4IOE5V9570.pdf),
```
Let's make this a list

frenck commented on source/_integrations/pi4ioe5v9xxxx.markdown:
```diff
@@ -0,0 +1,136 @@
+
+The `pi4ioe5v9xxxx` binary sensor platform allows you to read sensor values from the I/O pins of your I/O expander.
+
+The pin numbers are from 1 to X where: 1-8 correspond to port 0 (00-07) and 9-16 to port 1, etc.
```
```suggestion
The pin numbers are from 1 to X, where 1-8 correspond to port 0 (00-07) and 9-16 to port 1, etc.
```

frenck commented on source/_integrations/pi4ioe5v9xxxx.markdown:
```diff
@@ -0,0 +1,136 @@
+
+The `pi4ioe5v9xxxx` switch platform allows you to write to the I/O pins of your I2C I/O expander.
+
+The pin numbers are from 1 to X where: 1-8 correspond to port A (A1-A8) and 9-16 to port B (B1-B8), etc.
```
```suggestion
The pin numbers are from 1 to X, where 1-8 correspond to port A (A1-A8) and 9-16 to port B (B1-B8), etc.
```

frenck reviewed (changes requested):
Thanks for the PR @antonverburg!

Some small suggestions, could you please take a look? 👍

antonverburg committed:
Update source/_integrations/pi4ioe5v9xxxx.markdown

Co-Authored-By: Franck Nijhof <frenck@frenck.nl>

antonverburg committed:
Update pi4ioe5v9xxxx.markdown

antonverburg committed:
Update source/_integrations/pi4ioe5v9xxxx.markdown

Co-Authored-By: Franck Nijhof <frenck@frenck.nl>

antonverburg committed:
I2C bus number to optional

Modified bus from required to optional, as it will normally default to 1.

frenck reviewed (dismissed):
✅ Approved. Can be merged as soon as the parent PR gets merged.
⚠️ The `ha_release` might need an update before merging this one in.

antonverburg committed:
Merge pull request #1 from home-assistant/next

Update to latest version

antonverburg committed:
Create analog_output.markdown

Initial description of new analog-output entity, to be used to create new platforms.

antonverburg committed:
Added data attributed, fix typo

antonverburg committed:
Initial docu for slow_pwm

frenck reviewed (changes requested):
It seems like your latest changes have caused unrelated commits in this PR. Please clean up this PR so it contains only your commits.

Furthermore, please make sure the build passes. Thanks! 👍

antonverburg committed:
Delete analog_output.markdown

antonverburg committed:
Delete slow_pwm.markdown

antonverburg committed:
Update HA-release to 0.108

frenck committed:
:pencil2: Tweaks
